### ISSUE TITLE
onPress and onCalloutPress doesn't trigger on markers in iOS

### ISSUE THREAD

**Issue by gmarinosyan** [airbnb/react-native-maps#286]
Here is my code:

```
render() {
    return (
      <MapView style={styles.mapView} region={this.state.region}>
        <MapView.Marker
                coordinate={{latitude: 45, longitude: -75}}
                onSelect={this.markerPressed}
         />
      </MapView>
    )
}

markerPressed(){
    console.log("MarkerPressed");
}
```

The function is being called if I use onSelect or onDeselect, but not onPress or onCalloutPress. Is this an issue with something I am doing or a bug?

**Comment by christopherdro** [comment]
@gmarinosyan Is this for iOS or Android. onSelect and onDeselect if specifically for iOS and is what triggers the callout.

**Comment by gmarinosyan** [comment]
It is on iOS. Do the onPress and onCalloutPress functions not work on iOS?

**Comment by DennisMG** [comment]
@I'm experiencing the same problem... Did you solve it @gmarinosyan ??

**Comment by gmarinosyan** [comment]
No, was not able to solve.

**Comment by christopherdro** [comment]
@gmarinosyan Did you guys already try the most recent release 0.6.0?

**Comment by cjbutcher** [comment]
Think I'm having this issue on 0.6.0

**Comment by klaste** [comment]
@christopherdro I have the same problem. It seems as if the _handleTap gesture recognizer delegate method of AIRMapMarkerManager is never called. I am on 0.6.0, React Native 0.29-rc, and iOS 10, so I don't know if this is of any relevance or if I am looking in the right place anyway.

**Comment by klaste** [comment]
@christopherdro I will look into it. Do you have any idea what could be the cause?

**Comment by nickw** [comment]
@christopherdro @lelandrichardson I'm experiencing this issue as well using v0.6.0 and `react-native` v0.28.0. I'm happy to open a PR if anyone can point me in the right direction.

**Comment by KunwarSingh** [comment]
Hey man I'm also facing same issue, onSelect is working but not onCallbackPress. Please try to solve this issue.

**Comment by DmsChrisPena** [comment]
Yup, onPress isn't working at all for me on IOS. Any update on a work around or PR?

**Comment by KGALLET** [comment]
+1 onPress isn't working for me too on iOS.

**Comment by zelll** [comment]
@DmsChrisPena @KGALLET , I found this issue. It's a better place than #403 to answer your question.

See patch below. I'm not a iOS developer. Can someone review it and create a PR? @nickw ?

```
--- map2/node_modules/react-native-maps/ios/AirMaps/AIRMapMarker.m	2016-07-10 04:47:31.000000000 +0800
+++ map/node_modules/react-native-maps/ios/AirMaps/AIRMapMarker.m	2016-07-19 22:04:56.000000000 +0800
@@ -72,6 +72,10 @@
         if (_pinView == nil) {
             _pinView = [[MKPinAnnotationView alloc] initWithAnnotation:self reuseIdentifier: nil];
             _pinView.annotation = self;
+            _pinView.center = CGPointMake(16, 20); // make position/size of AIRMapMarker to match the pin
+            self.centerOffset = CGPointMake(-16, -20);
+            self.frame = CGRectMake(self.frame.origin.x, self.frame.origin.y, 32, 39);  // TODO: remove hard-coded size
+            [self addSubview:(UIView *) _pinView]; // add pin as child of marker and return marker itself
         }
 
         _pinView.draggable = self.draggable;
@@ -83,7 +87,7 @@
             _pinView.pinTintColor = self.pinColor;
         }
 
-        return _pinView;
+        return self;
     } else {
         // If it has subviews, it means we are wanting to render a custom marker with arbitrary react views.
         // if it has a non-null image, it means we want to render a custom marker with the image.
```

**Comment by yinghang** [comment]
You can fix the onPress issue by setting a width and height for the style of the marker.

**Comment by alsh76** [comment]
onCalloutPress still does not work for iOS for default callouts. it seems to be related to incorrect zero callout width. does anyone know a workaround for this?

**Comment by zjaml** [comment]
+1, onPress, onCalloutPress does not work for iOS.

**Comment by mahyarse** [comment]
+1, onPress does not work on iOS.
react-native: 0.32
react-native-maps: 0.8.0

**Comment by webgoudarzi** [comment]
+1, onPress does not work on iOS.
"react-native": "0.33.0",
"react-native-maps": "^0.8.2"

**Comment by monotv** [comment]
+1, onPress does not work on iOS.
"react-native": "^0.33.0",
"react-native-maps": "^0.8.2"

**Comment by ronnierios** [comment]
Still does no work on IOS
"react-native": "0.35.0",
"react-native-maps": "^0.10.2"

anybody could resolve?

**Comment by skylarmb** [comment]
+1

**Comment by shenders13** [comment]
+1 Still doesn't work on ios

**Comment by wilbyang** [comment]
RN 0.39
maps 0.12

still not working

**Comment by ippa** [comment]
Prob the same as #521 - see for some extra discussion

**Comment by wilbyang** [comment]
RN 0.39
maps 0.12

still not working.
if I copy the codes from example, it works
         ```
 <MapView.Marker
            title="This is a title"
            description="This is a description"
            coordinate={this.state.region}
            onPress={this.recordEvent('Marker::onPress')}
            onSelect={this.recordEvent('Marker::onSelect')}
            onDeselect={this.recordEvent('Marker::onDeselect')}
            onCalloutPress={this.recordEvent('Marker::onCalloutPress')}
          >
            <PriceMarker amount={99} />
            <MapView.Callout
              style={styles.callout}
              onPress={this.recordEvent('Callout::onPress')}
            >
              <View>
                <Text>Well hello there...</Text>
              </View>
            </MapView.Callout>
          </MapView.Marker>
```
But if I use a marker list and use map to generate the markers, it does not work


```
{this.props.events.map(event => {

              var latitude = parseFloat(event.latitude);
              var longitude = parseFloat(event.longitude);

              return (
                <MapView.Marker
                  key={event.index}
                  onPress={() => this.handleMapInteraction("markerPress")}
                  onSelect={() => this.handleMapInteraction("markerPress")}
                  onDeselect={() => this.handleMapInteraction("markerPress")}
                  onCalloutPress={() => this.handleMapInteraction("markerPress")}
                  coordinate={{latitude, longitude}}
                  title={event.company}
                  description={event.company}
                  calloutOffset={{ x: 0, y: 0 }}
                  calloutAnchor={{ x: 0.5, y: 0.4 }}
                ><PriceMarker amount={99} />
                  <MapView.Callout
                    style={styles.callout}
                    onPress={()=>this.handleMapInteraction('Callout::onPress')}
                  >
                    <View>
                      <Text>Well hello there...</Text>
                    </View>
                  </MapView.Callout>
                </MapView.Marker>
              )
            }
          )}
```

**Comment by tjdavies** [comment]
as onSelect only works on ios and onPress only works on android i'm using both as a work around
<MapView.Marker         
        onSelect={markerClickHandler}
        onPress={markerClickHandler}
    >

**Comment by pwoltman** [comment]
@RajkumarPunchh I can confirm that your commit has fixed the issue in my project - onCalloutPress behaves as expected.  Thanks!

**Comment by spikebrehm** [comment]
Fixed by #954

**Comment by OmgDef** [comment]
I came up with this
```javascript
<MapView.Callout tooltip={true} width={210} onPress={() => console.log(123)}>
````
